src/sql/command.py

```diff
@@ -89,3 +87,55 @@ def connection(self):
     def result_var(self):
         """Returns the result_var"""
         return self.parsed["result_var"]
+
+    def _params_helper(self, params):
+        def is_float(element):
+            if element is None:
+                return False
+            try:
+                float(element)
+                return True
+            except ValueError:
+                return False
+
+        params_dict = {}
+        if not params:
+            return params_dict
+            # Throw some exception
+
+        for key, value in params:
+            if is_float(value):
+                params_dict[key] = int(value)
+            else:
+                params_dict[key] = value
+        return params_dict
+
+    def _var_expand(self, magic, user_ns, line, cell):
+        """
+        Support for the variable substition in the SQL clause
+        For now, we have enabled two ways:
+        1. Latest format, {{a}}, we use jinja2 to parse the string with {{a}} format
+        2. Legacy format, {a}, $a, and :a format.
+
+        We will deprecate the legacy format feature in next major version
+        """
+        self.is_legacy_var_expand_parsed = False
+        # Latest format parsing
+        # TODO: support --param and --use-global logic here
+        line = Template(line).render(user_ns)
+        cell = Template(cell).render(user_ns)
+        # Legacy format parsing
+        parsed_cell = magic.shell.var_expand(cell, depth=2)
+        parsed_line = magic.shell.var_expand(line, depth=2)
+        # Exclusive the string with "://", but has :variable
+        has_SQLAlchemy_var_expand = re.search("(?<!://):[^/]+", line) or ":" in cell
+
+        if parsed_line != line or parsed_cell != cell or has_SQLAlchemy_var_expand:
+            self.is_legacy_var_expand_parsed = True
+            print(
+                "Please aware the variable substition"
+                " as {a}, $a, and :a format has been deprecated."
+            )
+            print("Use {{a}} instead.")
```

Comment:
also, let's add a tutorial in our docs describing this feature, and include the link to it

e.g., `https://jupysql.ploomber.io/en/latest/user-guide/templating.html`

https://github.com/ploomber/contributing/blob/main/documentation/notebooks.md

also in the current documentation where we explain the variable substitution, let's add a deprecation notice

~~~
```{versionchanged} 0.5.7
This has been deprecated use '{{a}}' [and add link to the new tutorial
```
~~~

Reply:
ah and one more thing: let's add a new document in the user guide where we list the incompatibilities with the ipython-sql project. just 1-2 sentences saying we use {{a}} instead of the old method and a link to the tutorial

Reply:
New template page in user-guide: https://jupysql--137.org.readthedocs.build/en/137/user-guide/template.html
Add deprecation notice in intro: https://jupysql--137.org.readthedocs.build/en/137/intro.html#variable-substitution

incompatibilities:
<img width="845" alt="Screenshot 2023-02-28 at 3 01 47 PM" src="https://user-images.githubusercontent.com/123580782/221965602-77277aa6-9c1c-49bb-87df-35d4b6943bc2.png">

Please ignore the latest build fail, since the new page `user-guide/template.html` haven't deployed to `jupysql.ploomber.io`, it won't pass the `broken-links` action in our CI

Reply:
> Looks like the CI is failing on broken links

Yes. this is expected, https://github.com/ploomber/jupysql/pull/137#discussion_r1120707722

Since we added the new page `user-guide/template.md`, and also referred it in another page ([vs.md](https://github.com/ploomber/jupysql/pull/137/files#diff-b02963516aaba6dc42cfae038f646149e604e224dc05635fd56e61023f9ff547)), but the template page haven't deployed to `jupysql.ploomber.io`

[6c4d9cb](https://github.com/ploomber/jupysql/pull/137/commits/6c4d9cb501d1ae2855373790debfa2b9167d30da) is a success build, since it it's without referring the template page. The latest commit is with that